### ISSUE TITLE
How to insert inline comments that explain what commands/remaps do in .neovintageousrc

### ISSUE THREAD

**Issue by eugenesvk** [NeoVintageous/NeoVintageous#864]
For example, I'd like to have
```
:noremap l h  "comment: Ⓝ l to ◀
:noremap ; l  "comment: Ⓝ ; to ▶
```

**Comment by gerardroche** [comment]
As far as I understand it's not possible to add regular comments after mapping commands in Vim "because the '"' character is considered to be part of the `{lhs}` or `{rhs}`", however, in Vim one can use |", since this starts a new, empty command with a comment.

```
							*map-comments*
It is not possible to put a comment after these commands, because the '"'
character is considered to be part of the {lhs} or {rhs}. However, one can
use |", since this starts a new, empty command with a comment.
```

It would be very tricky adding that feature i.e. allowing comments in the form of `|" ...` because as the docs point out, the pipe character is actually a chained command. The chained command in this case happens to be a comment.

There is currently very limited support for chained commands, see `:help nv-rc`:

> Chaining commands is supported by use of the special <Bar> character:
>
> ```
> nnoremap <leader>m :Cmd1<Bar>:Cmd2<CR>
> ```

It's enough to do most basic things you might want with chaining commands but comments won't work with the current implementation.

**Comment by eugenesvk** [comment]
Maybe a more general question - have you thought about using a much more ergonomic config format that overrides the vim format (e.g., leave the `.neovintageousrc` match the clunky vim format for compatibility where you can't even have comments, but have to pipe-construct some weird command for this functionality but then read another config that's better)
Like, this would also help with not having to repeat all those `noremap` you could have nested
```
:normal (or a list of modes for the bindings ↓)
  j move-line-down (much more readable vs. having just a single key and having to remember what it does in the original)	
  k move-line-up   
  l move-char-left	
  ; move-char-right re 
  (this would be ↑ recursive, just for illustration,
   but since nore is more common, it can be the default)

  (and then you can also have much more readable modifier+key maps)
  ⎈d scroll-lines-down instead of
  <C-d> (← would still be accepted as is currently, just ↑ would be an alternative) 
```

**Comment by gerardroche** [comment]
I think it's too much work to develop and maintain with not enough value in the end. Vim mappings get the job done for the most part. At a certain point people get their mappings to where they like 'em and then rarely change them. Maybe once in a blue moon you add or tweak them.

**Comment by eugenesvk** [comment]
>  At a certain point people get their mappings to where they like 'em and then rarely change them

Yes, but to get to that point you need to overcome a serious usability barrier, and every tweaking user has to do that, and the more you remap, the harder it is, so you have a builtin barrier to customization within a platform that's supposed to be very customizable
